Update board definitions for SC-OBC Module A1

sasataku opened the pull request:
In the EM (Engineering Model), the board was referred to as `Space Cubics Module 1`.
 Starting with the FM (Flight Model) release,  we are officially renaming it to:  `SC-OBC Module A1`
 Accordingly, the board name is updated from `scobc_module1` to  `scobc_a1`.

In addition to renaming the board to `SC-OBC Module A1`, the following changes are made:

- Updated UART IP core register addresses  
- Modified HRMEM (SRAM) size  
- Changed the number of IRQs  
- Updated FPGA System Register / System Monitor Register address to DTS

These changes are based on the specifications described in the following manual:
https://sc-obc-fpga-technical-reference-manual.readthedocs.io/ja/latest/

And we also decided to introduce a Revision, setting the EM to `1.0.0`, the FM to `2.0.0`, and using 2.0.0 as the default.

sasataku committed:
boards: scobc_a1: Rename board name to scobc-a1

In the EM (Engineering Model), the board was referred to as
"Space Cubics Module 1". Starting with the FM (Flight Model) release,
we are officially renaming it to:

  `SC-OBC Module A1`

Accordingly, the board name is updated from `scobc_module1` to
`scobc_a1`.

Signed-off-by: Takuya Sasaki <takuya.sasaki@spacecubics.com>

sasataku committed:
doc: scobc_a1: Rename board name

Rename the board name to `scobc-a1`, and update the board's full name
to `SC-OBC Module A1`. And update the output of the `Hello World`
sample to match the latest Zephyr version.
Then, The supported feature section was replaced with a hw-directive
block.

Signed-off-by: Takuya Sasaki <takuya.sasaki@spacecubics.com>

sasataku committed:
dts: scobc_a1: Add FPGA related definitions

Adds SC-OBC Module A1 FPGA System Register and System Monitor
definitions to DTS.

  - System Register
    Registers for system control of SC-OBC Module A1
    Link: https://sc-obc-fpga-technical-reference-manual.readthedocs.io/ja/latest/#org19067cd

  - System Monitor
    Module for monitoring the SC-OBC Module A1 system
    Link: https://sc-obc-fpga-technical-reference-manual.readthedocs.io/ja/latest/#org8832978

Signed-off-by: Takuya Sasaki <takuya.sasaki@spacecubics.com>

sasataku committed:
boards: scobc_a1: Increase number of IRQs

Increase number of IRQs for SC-OBC Module A1 to 32, including
reserved entries.

Signed-off-by: Takuya Sasaki <takuya.sasaki@spacecubics.com>

nordicjm commented on boards/sc/scobc_a1/Kconfig.defconfig:
typo in commit title *rename

nordicjm reviewed (commented):
should add a deprecated board entry in `boards/deprecated.cmake`

sasataku committed:
boards: scobc_a1: Change HRMEM and UART definitions

The following changes have been made in the FM (Flight Model) of the
`SC-OBC Module A1`:

  - Change in the UART Core register address
    Link: https://sc-obc-fpga-technical-reference-manual.readthedocs.io/ja/latest/#orgecb2e01

  - Change in the size of HRMEM (High-Reliability Memory)
    Link: https://sc-obc-fpga-technical-reference-manual.readthedocs.io/ja/latest/#orgc3b1ecc

As a result, we have defined the EM board revision as `1.0.0` and the
FM as `2.0.0`. The EM configuration has been moved to
`scobc_a1_1_0_0.overlay`.

Signed-off-by: Takuya Sasaki <takuya.sasaki@spacecubics.com>

sasataku committed:
boards: scobc_a1: Fix YAMLLint error

Fixes the YAMLLint error as below:

   YAMLLint (indentation):wrong indentation: expected 4 but found 2
   File:boards/sc/scobc_a1/board.yml
   Line:12
   Column:3

Signed-off-by: Takuya Sasaki <takuya.sasaki@spacecubics.com>

nordicjm commented on boards/sc/scobc_a1/doc/index.rst:
```diff
@@ -21,7 +21,7 @@ Hardware
-The Space Cubics OBC module 1 provides the following hardware features:
+The SC-OBC Module A1 provides the following hardware features:
 
 +-----------+------------+------------------------------------+
```
n.b. this section should be replaced with the hw-directive block instead (for a future PR)

sasataku replied:
Thanks a lot!  I looked at the other boards and gave it a try in the hw-directive block.

nordicjm commented on boards/sc/scobc_a1/doc/index.rst:
```diff
@@ -35,7 +35,7 @@ The Space Cubics OBC module 1 provides the following hardware features:
 
 The default configuration for the board can be found in the defconfig file:
-:file:`boards/arm/scobc_module1/scobc_module1_defconfig`.
+:file:`boards/arm/scobc_a1/scobc_a1_defconfig`.
```
path does not exist, so is probably easier to replace section with hw-directive block

kartben replied:
yes +1, the board doc page already has a direct link to "Browse board sources" which should be more than enough

sasataku replied:
Yes, I replaced. Thanks!
